### ISSUE TITLE
The security problem about cs_reregister

### ISSUE THREAD

**Issue by elct9620** [unlightcpa/Unlight#12]
I have some question about this code

https://github.com/unlightcpa/Unlight/blob/015d92e4493f81a8e7975f60f7473e156322187b/app/server/src/protocol/authserver.rb#L144-L148

The SRP didn't save the user's password on the server, but this command allows anyone to change the player's password without any check.

This should be protected or disabled if the server is using the username and password to log in, but I seem not find the related protected in the source code.

**Comment by Karube** [comment]
Ignore this code. This is because it is a code for fallback in case of password saving failure on the SNS side.

**Comment by elct9620** [comment]
I think it should be able to disable when we are using the normal username and password login. When I test the login function, it can be called without any check on the server. If I know the server address and username, it can change any user's password.

**Comment by Karube** [comment]
You're right. This code was executed only for users who were logged in via SNS, so it could be changed if the unique ID was known.
It is completely a security issue in situations where IDs can be created freely. I think that it should be deleted from commands in actual operation.

**Comment by elct9620** [comment]
Maybe we can add a feature flag to toggle it on/off or add simple protected with verifying a secret token to prevent any user can send the command to the server.

I think this feature will be used after people starting to integrate their website or social network.
